plugins/source_scheduler_dfs.go

```diff
@@ -36,6 +36,16 @@ func (p *SourcePlugin) syncDfs(ctx context.Context, spec specs.Source, client sc
 	}
 	p.resourceSem = semaphore.NewWeighted(int64(resourceConcurrency))
 
+	for _, table := range tables {
+		clients := []schema.ClientMeta{client}
+		if table.Multiplex != nil {
+			clients = table.Multiplex(client)
+		}
+		// we do this here to avoid locks so we initial the metrics structure once in the main goroutines
+		// and then we can just read from it in the other goroutines concurrently given we are not writing to it.
+		p.metrics.initWithClients(table, clients)
```

Comment:
`initWithClients` recursively traverses the tables, so this still has the (minor) issue that some tables will be initialized twice. Maybe let's add `if table.Parent != nil {` to the top of this loop for now so that it's correct, and we can clean up/refactor later?

Reply:
oh didn't catch that! Yes, definitely only init parent tables as children will be handled by the initWithClients call.